GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1473,8 +1476,8 @@ bool IsCommitMessageValid()
                     {
                         try
                         {
-                            if (!Message.Text.StartsWith(fixupPrefix) &&
-                                !Message.Text.StartsWith(squashPrefix) &&
+                            if (!Message.Text.StartsWith(CommitKind.Fixup.GetPrefix()) &&
+                                !Message.Text.StartsWith(CommitKind.Squash.GetPrefix()) &&
                                 !Regex.IsMatch(Message.Text, AppSettings.CommitValidationRegEx) &&
                                 MessageBox.Show(this, _commitMsgRegExNotMatched.Text, _commitValidationCaption.Text, MessageBoxButtons.YesNo, MessageBoxIcon.Asterisk) == DialogResult.No)
```

Comment:
I am not sure about this. The text in the dialog would contain not just the edited commit message, but also the "amend header", that is the first line will be `amend! <original commit subject>` which might throw the validation off. But perhaps that should be the responsibility of the users to take the "amend header" into account in the validation regex?

Reply:
"amend!" messages should simply be excluded from the Regex check or rather the amend header should be removed before the check, so that the resulting message is validated.